solr/core/src/java/org/apache/solr/cloud/overseer/ZkStateWriter.java

```diff
@@ -113,20 +114,51 @@ public ClusterState enqueueUpdate(ClusterState prevState, List<ZkWriteCommand> c
     if (cmds.isEmpty()) return prevState;
     if (isNoOps(cmds)) return prevState;
 
+    boolean forceFlush = false;
+    if(cmds.size() == 1) {
+      //most messages result in only one command. let's deal with it right away
+      ZkWriteCommand cmd = cmds.get(0);
+      if (cmd.collection != null && cmd.collection.isPerReplicaState()) {
+        //we do not wish to batch any updates for collections with per-replica state because
+        // these changes go to individual ZK nodes and there is zero advantage to batching
+        //now check if there are any updates for the same collection already present
+        if (updates.containsKey(cmd.name)) {
+          //this should not happen
```

Comment:
Do we want to log this somehow if it's removing things? That sounds like it could be a subtle bug to find later if it somehow did encounter this.

Reply:
It's not removing , it's just taking them out and executing right away